src/PhpPact/Standalone/PactConfig.php

```diff
@@ -0,0 +1,219 @@
+     * Consumer name.
+     *
+     * @var string
+     */
+    private string $consumer;
+
+    /**
+     * Provider name.
+     *
+     * @var string
+     */
+    private string $provider;
+
+    /**
+     * Directory to which the pacts will be written.
+     *
+     * @var null|string
+     */
+    private ?string $pactDir = null;
+
+    /**
+     * The pact specification version to use when writing the pact. Note that only versions 1, 2, 3 and 4 are currently supported.
+     *
+     * @var string
+     */
+    private string $pactSpecificationVersion;
+
+    /**
+     * File to which to log output.
+     *
+     * @var null|string
+     */
+    private ?string $log = null;
+
+    /**
+     * @var null|string
+     */
+    private ?string $logLevel = null;
+
+    /**
+     * `overwrite` or `merge`. Use `merge` when running multiple mock service
+     * instances in parallel for the same consumer/provider pair. Ensure the
+     * pact file is deleted before running tests when using this option so that
+     * interactions deleted from the code are not maintained in the file.
+     *
+     * @var string
+     */
+    private string $pactFileWriteMode = self::MODE_OVERWRITE;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConsumer(): string
+    {
+        return $this->consumer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setConsumer(string $consumer): self
+    {
+        $this->consumer = $consumer;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getProvider(): string
+    {
+        return $this->provider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setProvider(string $provider): self
+    {
+        $this->provider = $provider;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPactDir(): string
+    {
+        if ($this->pactDir === null) {
+            return \sys_get_temp_dir();
+        }
+
+        return $this->pactDir;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setPactDir(?string $pactDir): self
+    {
+        if ($pactDir === null) {
+            return $this;
+        }
+
+        if ('\\' !== \DIRECTORY_SEPARATOR) {
+            $pactDir = \str_replace('\\', \DIRECTORY_SEPARATOR, $pactDir);
+        }
+
+        $this->pactDir = $pactDir;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPactSpecificationVersion(): string
+    {
+        return $this->pactSpecificationVersion;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \UnexpectedValueException
+     */
+    public function setPactSpecificationVersion(string $pactSpecificationVersion): self
+    {
+        /*
+         * Parse the version but do not assign it.  If it is an invalid version, an exception is thrown
+         */
+        $parser = new VersionParser();
+        $parser->normalize($pactSpecificationVersion);
+
```

Comment:
I don't think using `semver` here is necessary.  I use hard coded versions above to check again anyway. So I think we can remove `composer/semver` and this code.

```suggestion
```

See https://github.com/pact-foundation/pact-php/pull/291

Reply:
Resolve to make code reviewing easier. Feel free to reopen or add comment.

Reply:
Reverted removing code.